### DISCUSSION TITLE
I can introduce the elemnt but I can't  show if a elemt in the list     voidbuscarLista(Nodo *lista, int n) help

### DISCUSSION THREAD

**Discussion by 19Alvaro** [community/community#69065] [Programming Help]
include<iostream>
#include<conio.h>

using namespace std;

struct Nodo{       //primer paso
	int dato;
	Nodo *siguiente;
};

void insertarLista(Nodo *&,int);

void mostrarLista(Nodo *lista);
void buscarLista(Nodo *lista, int n );

int main(){
	Nodo *lista = NULL;  //creamos nuestro puntero
	int dato;
	int n;
	
	cout<<"Introduce un numero:";
	cin>>dato;
	insertarLista(lista,dato);
	
	cout<<"Introduce un numero:";
	cin>>dato;
	insertarLista(lista,dato);
	
	cout<<"Introduce un numero:";
	cin>>dato;
	insertarLista(lista,dato);
	
	cout<<"Introduce un numero:";
	cin>>dato;
	insertarLista(lista,dato);
	
	cout<<"Introduce un numero:";
	cin>>dato;
	insertarLista(lista,dato);
	
	cout<<"Introduce un numero:";
	cin>>dato;
	insertarLista(lista,dato);
	
	cout<<"Que numero quieres buscar:";
	cin>> n;
	
	
	mostrarLista(lista);
	
	getch();
	return 0;
	
}

void insertarLista(Nodo *&lista,int n){
	Nodo *nuevo_nodo = new Nodo(); //creamos nuevo nodo 
	nuevo_nodo ->dato = n;
	
	Nodo *aux1 = lista;
	Nodo *aux2;
	
	while((aux1 != NULL) && (aux1 ->dato < n)){
		aux2 = aux1;
		aux1 = aux1 ->siguiente;
	}
	if(lista == aux1){
		lista = nuevo_nodo;
	}
	else{
		aux2 ->siguiente = nuevo_nodo;
	}
	nuevo_nodo->siguiente = aux1;
	cout<<"\tElemento "<<n<<"insertado a lista correctamente\n";
	
}
void mostrarLista(Nodo *lista){
	Nodo *actual = new Nodo();
	actual = lista;
	
	while(actual != NULL){
		cout<<actual ->dato<<"->";
		actual = actual ->siguiente;
		
	}
}
void buscarLista(Nodo *lista,int n){
	bool band = false;
	
	Nodo *actual = new Nodo();
	actual = lista;
	
	while((actual != NULL) && (actual ->dato <= n)){
		if(actual->dato == n){
			band = true;
		}
		actual = actual ->siguiente;
		
	}
	if(band == true){
		cout<<"Elemento " <<n<< " Si a sido encontrado en lista\n";
	}
	else{
		cout<<"Elemento" <<n<< " No a sido encontrado en la lista\n";
	}
}

**Comment by Abkeyev99** [comment]
Thanks so good :)

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @19Alvaro ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by 19Alvaro** [reply]
Thanks

**Comment by jamesweber7** [comment]
Me parece que buscarLista está funcionando correctamente. Después de obtener el número que quieres buscar, estás llamando a mostrarLista en lugar de a buscarLista. Intenta esto en su lugar:
```
cout << "Que numero quieres buscar:";
cin >> n;

// mostrarLista(lista);
buscarLista(lista, n);
```

**Reply by 19Alvaro** [reply]
Gracias por tu atención y ayuda saludos
